synapse/state/v2.py

```diff
@@ -430,7 +430,7 @@ async def _iterative_auth_checks(
     Returns:
         Returns the final updated state
     """
-    resolved_state = base_state.copy()
+    resolved_state = dict(base_state)
```

Comment:
`MutableMapping` doesn't have a `copy()` method (although `Dict` does). This seemed equivalent.

Reply:
I agree — both do shallow copies.